app-frontend/src/app/components/diagramContainer/diagramContainer.controller.js

```diff
@@ -17,82 +20,177 @@ export default class DiagramContainerController {
-                height: 75
+                width: this.cellSize[0],
+                height: this.cellSize[1]
             },
             attrs: {
                 rect: {
-                    rx: 4,
-                    ry: 7,
-                    fill: '#dfdfdf'
+                    fill: '#f8f9fa'
                 },
                 text: {
                     fill: '#333333',
-                    text: this.cellLabel
+                    text: label
                 }
             },
             ports: {
                 groups: {
                     inputs: {
                         position: {
-                            name: 'top'
+                            name: 'left'
                         }
                     },
-                    functions: {
+                    outputs: {
                         position: {
-                            name: 'left'
+                            name: 'right'
                         }
                     }
                 },
-                items: [
-                    {
-                        group: 'inputs'
-                    },
-                    {
-                        group: 'functions'
-                    }
-                ]
+                items: this.createPorts(config.inputs, config.outputs)
             }
-        }));
+        });
+
+        this.shapes.push(shape);
+        return shape;
     }
 
-    initDiagram() {
-        this.graph = new joint.dia.Graph();
-        this.paper = new joint.dia.Paper({
-            el: this.workspaceElement,
-            height: $(this.workspaceElement).height(),
-            width: $(this.workspaceElement).width(),
-            gridSize: 25,
-            drawGrid: true,
-            model: this.graph
+    createPorts(inputs, outputs) {
+        let ports = [];
+        inputs.forEach(i => {
+            ports.push({
+                id: i,
+                group: 'inputs'
+            });
         });
-        this.paper.drawGrid({
-            color: '#aaa',
-            thickness: 1
+        outputs.forEach(o => {
+            ports.push({
+                id: o,
+                group: 'outputs'
+            });
         });
-        this.paper.on('blank:pointerclick', () => {
-            if (this.onPaperClick) {
-                this.onPaperClick();
+
+        return ports;
+    }
+
+    createLink(src, target) {
+        let link = new joint.dia.Link({
+            source: {id: src[0], port: src[1]},
+            target: {id: target[0], port: target[1]},
+            attrs: {
+                '.marker-target': {
+                    d: 'M 4 0 L 0 2 L 4 4 z'
+                }
             }
         });
-        this.paper.on('cell:pointerclick', () => {
-            if (this.onCellClick) {
-                this.onCellClick();
-            }
+        this.shapes.push(link);
+        return link;
+    }
+
+    centerGraph() {
+        let bbox = this.graph.getBBox();
+        this.graph.set({
```

Comment:
This doesn't seem to actually be centered on the page 

![screenshot from 2017-01-12 14 57 46](https://cloud.githubusercontent.com/assets/898060/21906040/a52b1fcc-d8d7-11e6-9b0e-918fa6564757.png)